chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/CertifiedProductController.java

```diff
@@ -370,9 +372,18 @@ public CertifiedProductController(final CertifiedProductUploadHandlerFactory upl
 
         certifiedProduct = validateCertifiedProduct(certifiedProduct);
 
+        removeNonBasicCertifiedProductSearchDetails(certifiedProduct);
+
         return certifiedProduct;
```

Comment:
If we're still returning the same type of object, won't it still have the keys that we want removed?

Reply:
No, because of the Jackson annotation on the DTO tells the mapper to exclude the class field members that are null.